app/views/handlers/submission.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+import simplejson as json
+from flask import url_for, current_app
+from sdc.crypto.encrypter import encrypt
+from app.globals import get_session_store
+from app.keys import KEY_PURPOSE_SUBMISSION
+from app.submitter.converter import convert_answers
+from app.submitter.submission_failed import SubmissionFailedException
+
+
+class SubmissionHandler:
+    def __init__(self, schema, questionnaire_store, full_routing_path):
+        self.schema = schema
+        self.questionnaire_store = questionnaire_store
+        self.full_routing_path = full_routing_path
+
+    def submit_message(self):
+        message = json.dumps(
+            convert_answers(
+                self.schema, self.questionnaire_store, self.full_routing_path
+            ),
+            for_json=True,
+        )
```

Comment:
Given that the schema and routing_path are only used to generate the message, it feels like a more suitable interface might be for the handler to be initialised only with metadata and handler.submit_message() to take the message as an arg, rather than fetch it.

Reply:
You're right, that does make sense. The handler shouldn't be going to get the message. Thanks

Reply:
I don't think so, as we're trying to move logic away from the questionnaire routes. Maybe it would be clearer if this was named `submit_questionnaire`?